compiler/passes/scopeResolve.cpp

```diff
@@ -1548,6 +1548,23 @@ static void resolveUnresolvedSymExpr(UnresolvedSymExpr* unresolvedSymExpr,
 
           unresolvedSymExpr->replace(prim_capture_fn);
           prim_capture_fn->insertAtTail(unresolvedSymExpr);
+          //
+          // If we detect that this function reference is within a
+          // c_fnPtrTo() call then we only need a C pointer to the
+          // function, not a full Chapel first-class function (which
+          // can capture variables).  We'll mark this so that function
+          // resolution can distinguish by adding a second sentinel
+          // argument to the PRIM_CAPTURE_FN call ('true',
+          // arbitrarily).  We could alternatively add another
+          // primitive to distinguish the C case, but for most
+          // intents, it behaves identically to PRIM_CAPTURE_FN, so
+          // this seemed like a waste of a primitive...
```

Comment:
OK, I can do that if you guys prefer.  I actually did this at first, but the number of places in the compiler that I had to touch (not a ton, perhaps a half dozen) seemed annoying given how small the differences between them were (exit early in one place).  So I pulled all the code out and fell back to this approach instead.

Another idea I was toying with was whether we could eliminate the primitive altogether, use c_fnPtrTo() in its place (for both the C and Chapel cases), but to mark the Chapel case in some way that marks it as Chapel such that it can get rewritten and pulled out (where today, the primitive is rewritten and pulled out).  Advantages: fewer primitives, more externs; but probably still something subtle to distinguish the two cases (but at least we have the advantage of potentially seeing a named argument or something like that).